### PR TITLE
allow swagger endpoints to be disabled via configuration item.

### PR DESCRIPTION
**Pull request by rolfyone**
added enableSwagger to config.toml, and defaulted to true.

When enableSwagger is set to false, both swagger-docs and swagger-ui become unavailable.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

#1227

### CODE REVIEW

File: data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/BeaconRestApiTest.java [hunk 1]
```diff
@@ -41,12 +42,15 @@
       mock(CombinedChainDataClient.class);
   private final JavalinServer server = mock(JavalinServer.class);
   private final Javalin app = mock(Javalin.class);
-  private static final Integer THE_PORT = 12345;
+  private static final Integer THE_PORT = 5051;
```
Comment (macfarla) [inline data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/BeaconRestApiTest.java]: ohhh 12345 had a certain ring to it

File: data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/BeaconRestApi.java [hunk 2]
```diff
@@ -132,14 +135,10 @@ private static OpenApiOptions getOpenApiOptions(JsonProvider jsonProvider) {
...
                     .url("https://www.apache.org/licenses/LICENSE-2.0.html"));
-    OpenApiOptions options =
-        new OpenApiOptions(applicationInfo)
-            //            .jacksonMapper(factory.getObjectMapper())
-            .modelConverterFactory(factory)
-            .path("/swagger-docs")
-            .swagger(new SwaggerOptions("/swagger-ui"));
-    // TODO: allow swagger-ui to be turned off - ideally still leave swagger-docs, just dont add the
-    // swagger-ui endpoint
+    OpenApiOptions options = new OpenApiOptions(applicationInfo).modelConverterFactory(factory);
```
Comment (macfarla) [inline data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/BeaconRestApi.java]: final

File: data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/BeaconRestApiTest.java [hunk 3]
```diff
@@ -41,12 +42,15 @@
...
 
   @BeforeEach
   public void setup() {
+    ArtemisConfiguration config =
+        ArtemisConfiguration.fromString(
+            "beaconrestapi.portNumber=5051\nbeaconrestapi.enableSwagger=false");
```
Comment (macfarla) [inline data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/BeaconRestApiTest.java]: can 5051 not be hardcoded in here

Comment (rolfyone) [reply data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/BeaconRestApiTest.java]: it's just configuring the configuration object for the test suite... alternative?

**Review by macfarla — APPROVED** [review]

**Comment by rolfyone** [comment]
[beaconrestapi]
portNumber = 5051
**enableSwagger = true**

**Pull request merged**